packages/databricks-vscode/resources/python/00-databricks-init.py

```diff
@@ -357,6 +359,104 @@ def df_html(df):
     html_formatter.for_type(SparkConnectDataframe, df_html)
     html_formatter.for_type(DataFrame, df_html)
 
+@logErrorAndContinue
+@disposable
+def register_spark_progress(spark):
+    try:
+        from pyspark.sql.connect.shell.progress import Progress
+        import ipywidgets as widgets
+    except Exception as e:
+        return
+
+    class Progress:
+        SI_BYTE_SIZES = (1 << 60, 1 << 50, 1 << 40, 1 << 30, 1 << 20, 1 << 10, 1)
+        SI_BYTE_SUFFIXES = ("EiB", "PiB", "TiB", "GiB", "MiB", "KiB", "B")
+
+        def __init__(
+            self,
+        ) -> None:
+            self._ticks = None
+            self._tick = None
+            self._started = time.time()
+            self._bytes_read = 0
+            self._running = 0
+            self.init_ui()
+
+        def init_ui(self):
+            self.w_progress = widgets.IntProgress(
+                value=0,
+                min=0,
+                max=100,
+                bar_style='success',
+                orientation='horizontal'
+            )
+            self.w_status = widgets.Label(value="")
+            if self.is_enabled():
+                display(widgets.HBox([self.w_progress, self.w_status]))
+
+        def is_enabled(self):
+            env_var = os.getenv("DATABRICKS_CONNECT_PROGRESS")
```

Comment:
We have a `LocalDatabricksNotebookConfig` object that loads environment and populates member variables similar to the way SDK configs work. Better to use that so that we have all the configs collected in 1 place.

Reply:
Thanks. That's what I need